core-patch/src/main/java/eu/neverblink/jelly/core/patch/internal/PatchDecoderImpl.java

```diff
@@ -0,0 +1,355 @@
+package eu.neverblink.jelly.core.patch.internal;
+
+import eu.neverblink.jelly.core.JellyOptions;
+import eu.neverblink.jelly.core.ProtoDecoderConverter;
+import eu.neverblink.jelly.core.RdfProtoDeserializationError;
+import eu.neverblink.jelly.core.RdfTerm;
+import eu.neverblink.jelly.core.internal.DecoderBase;
+import eu.neverblink.jelly.core.patch.JellyPatchOptions;
+import eu.neverblink.jelly.core.patch.PatchDecoder;
+import eu.neverblink.jelly.core.patch.PatchHandler;
+import eu.neverblink.jelly.core.proto.v1.*;
+
+public abstract class PatchDecoderImpl<TNode, TDatatype> extends DecoderBase<TNode, TDatatype> implements PatchDecoder {
```

Comment:
Please re-mark everything related to Jelly-Patch as experimental. There is an annotation in protobuf java called @ExperimentalApi – maybe this?